src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -119,7 +122,13 @@ void Convolution<
 
   if (padW != 0 || padH != 0)
   {
-    Pad(inputTemp, padW, padH, inputPaddedTemp);
+    inputPaddedTemp.set_size(inputTemp.n_rows + padW * 2,
+        inputTemp.n_cols + padH * 2, inputTemp.n_slices);
+    for (size_t i = 0; i < inputTemp.n_slices; ++i)
+    {
+      boost::apply_visitor(ForwardVisitor(std::move(inputTemp.slice(i)),
```

Comment:
If we use the ``Padding`` type, see comment above, we could just call ``padding.Forward(..)`` without the visitor, let me know what you think.